### PR TITLE
fallback address for recieving royalties

### LINKED ISSUE #8512: Check if user has at least existential deposit when royalty is paid
### Is your feature request related to a problem?

When you will try to send royalty in #8511 to user that has no balance. The transaction is will fail therefore before sending an royalty check if user has balance.

### The transaction is will fail

Token: Below Minimum error:

image

https://assethub-polkadot.subscan.io/extrinsic/5229366-2

https://assethub-polkadot.subscan.io/account/1L2xstToYKYE3m7R2qLM8MMpGQNmioSkMpx39QcxQucgnNZ

### Describe the solution you would like

Algo:

if user has more than ED. Do it
if user has less than ED but your royalty makes it alive. Do it
if user has less than ED and your royalty won't make it send it to the specified address (will provide).

### Describe alternatives you have considered

- 
Maybe there is make alive function.

### Screenshots

_No response_

### Are there opened related issues?

_No response_

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [x] Feature

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #8512

#### Did your issue had any of the "$" label on it?

$

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

### CODE REVIEW

**Comment by daiagi** [comment]
tested with the following setup
let's mark 4 account by name
creator account (creates the nfts)(C)
buyer account(B)
royalty receiving account (RR)
Royalty Fallback account (RF)

#### Test 1
create an nft and buy it, while RR account does not have enough money in it
expected result , royalty land in RF account

:heavy_check_mark:  Success
see nft details and description: /ahp/gallery/53-1
and transaction: https://assethub-polkadot.subscan.io/extrinsic/5243039-2

#### Test 2
top up RR account with 1 dot
mint and nft and buy it
expected result, royalty lands in RR account
:heavy_check_mark:  Success

see nft details and description: /ahp/gallery/53-2
and transaction: https://assethub-polkadot.subscan.io/extrinsic/5243076-2

File: composables/transaction/transactionBuy.ts [hunk 1]
```diff
@@ -9,8 +9,27 @@ import { getApiCall } from '@/utils/gallery/abstractCalls'
...
+}
+
+const getFallbackAddress = () => {
+  // subtrate format
+  // TODO:
+  //  1.change to real royalty recipient
+  //  2. move to static?
+
+  const FALBACK_ROYALTY_RECIPIENT =
+    '5EUDt8Gu6e1tQozggApnHbzFTqwGwZs7696Hupk595XTa2Ht'
```
Comment (vikiival) [inline composables/transaction/transactionBuy.ts:28]: Whose address is this?

Please use different one 
Cc @yangwao

File: composables/transaction/transactionBuy.ts [hunk 2]
```diff
@@ -9,8 +9,27 @@ import { getApiCall } from '@/utils/gallery/abstractCalls'
 import { payRoyaltyTx, somePercentFromTX } from '@/utils/support'
 import type { ActionBuy } from './types'
 import { verifyRoyalty } from './utils'
+import { existentialDeposit } from '@kodadot1/static'
+import { decodeAddress, encodeAddress } from '@polkadot/util-crypto'
+const formatAddress = (address: string): string => {
+  const { chainProperties } = useChain()
 
-function payRoyaltyAssetHub(
+  const publicKey = decodeAddress(address)
```
Comment (vikiival) [inline composables/transaction/transactionBuy.ts:17]: , you can directly pass address into encode. 

+ we have already utils for that in kodadot

File: composables/transaction/transactionBuy.ts [hunk 3]
```diff
@@ -20,21 +39,40 @@ function payRoyaltyAssetHub(
...
     return
   }
 
+  const balanceOfRoyaltyReceiver = await getNativeBalance({
+    address: normalizedRoyalty.address,
+    api,
+  })
+  const royaltyAmount = royaltyFee(price, normalizedRoyalty.amount)
+
+  const accountBalanceWithRoyalty =
```
Comment (vikiival) [inline composables/transaction/transactionBuy.ts:54]: Number can fail on bigger amouts (eg 3000 DOT ) prefer bigint

File: composables/transaction/transactionBuy.ts (hunk 1 above)
Comment (daiagi) [reply composables/transaction/transactionBuy.ts:28]: It's mine.
Obviously use different one
Lmk which

Comment (daiagi) [reply composables/transaction/transactionBuy.ts:28]: It's mine.
Obviously use different one
Lmk which

File: composables/transaction/transactionBuy.ts (hunk 2 above)
Comment (daiagi) [reply composables/transaction/transactionBuy.ts]: >you can directly pass address into encode.

done

>we have already utils for that in kodadot

didn't find something that was more convenient to use then `encodeAddress` directly

File: composables/transaction/transactionBuy.ts (hunk 3 above)
Comment (daiagi) [reply composables/transaction/transactionBuy.ts]: done

File: composables/transaction/transactionBuy.ts (hunk 1 above)
Comment (daiagi) [reply composables/transaction/transactionBuy.ts:24]: Sorry, could not find any of these

I am familiar with Wayland fund and kodadot minter

https://polkadot.subscan.io/account/15CoYMEnJhhWHvdEPXDuTBnZKXwrJzMQdcMwcHGsVx5kXYvW


Mind providing an address?

Comment (vikiival) [reply composables/transaction/transactionBuy.ts:24]: All if defined here `utils/support.ts`

**Review by daiagi — COMMENTED** [review]
@vikiival done

**Review by preschian — APPROVED** [review]

File: composables/transaction/transactionBuy.ts [hunk 4]
```diff
@@ -20,21 +32,42 @@ function payRoyaltyAssetHub(
 ) {
   const { isValid, normalizedRoyalty } = verifyRoyalty(royalty)
 
+  const { urlPrefix } = usePrefix()
```
Comment (roiLeo) [inline composables/transaction/transactionBuy.ts:35]: already in `item` inside `execBuyStatemine`, just pass the whole item

**Review by roiLeo — COMMENTED** [review]
mixed arrow function/nomal function

otherwise lgtm

**Review by vikiival — APPROVED** [review]
LGTM needs testing,

What if the royalty is empty?

**Comment by daiagi** [comment]
> LGTM needs testing,
> 
> What if the royalty is empty?

It's being checked, as it was checked before this pr

**Comment by yangwao** [comment]
lgtm, 
tests are failing, cc @prury

**Comment by yangwao** [comment]
Thanks!
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 7.08 USD/DOT ~ 4.237 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0xe2e5c74a2bcdcdac2495221e9e9e11f796e8165c1558c9c2384d5ddbb44bbac0](https://polkadot.subscan.io/extrinsic/0xe2e5c74a2bcdcdac2495221e9e9e11f796e8165c1558c9c2384d5ddbb44bbac0)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**